### PR TITLE
Refactorization

### PR DESCRIPTION
**Pull request by kamiboers**
- remove redundant score method
- refactor ball class, board class and cube drawing method (cube class) to be more readable
- refactor index.js initial save onclick method

All game function is maintained, all unit tests are still passing.

### CODE REVIEW

File: lib/ball.js [hunk 1]
```diff
@@ -17,38 +17,64 @@ function Ball(params) {
...
-    this.currentPosition = this.nextPosition;
+    this.isInSitu();
   }
 };
 
 Ball.prototype.getYPosition = function() {
   return this.onCube.y;
 };
 
+Ball.prototype.jumpIntervalElapsed = function(tick) {
```
Comment (s-espinosa) [inline lib/ball.js]: Maybe call this 'timeToMove'?

File: lib/ball.js [hunk 2]
```diff
@@ -17,38 +17,64 @@ function Ball(params) {
...
+  return tick % this.moveInterval === 0;
+};
+
+Ball.prototype.jumpingOffBoard = function() {
+  return this.nextPosition === null && this.y < 620;
+};
+
+Ball.prototype.leavingScreen = function() {
```
Comment (s-espinosa) [inline lib/ball.js]: 'leftScreen'? Also, does JS accept '?' in method names? Might make this even more readable.

File: lib/ball.js [hunk 3]
```diff
@@ -17,38 +17,64 @@ function Ball(params) {
...
+  return this.nextPosition === null && this.y > 620;
+};
+
+Ball.prototype.setVelocity = function(x, y) {
```
Comment (s-espinosa) [inline lib/ball.js]: Think this is setPosition, right?

File: lib/ball.js [hunk 4]
```diff
@@ -17,38 +17,64 @@ function Ball(params) {
 }
 
 Ball.prototype.update = function(tick) {
-  if(tick % this.moveInterval === 0){
+  if(this.jumpIntervalElapsed(tick)){
```
Comment (s-espinosa) [inline lib/ball.js]: Love these new methods as a way to refactor.

File: lib/ball.js [hunk 5]
```diff
@@ -17,38 +17,64 @@ function Ball(params) {
...
+
+Ball.prototype.inTransit = function() {
+  return this.x !== this.targetX;
+};
+
 Ball.prototype.onCube = function(){
   return this.board.cubes[this.currentPosition] || this.board.cubes[0];
 };
 
+Ball.prototype.isInSitu = function(){
```
Comment (s-espinosa) [inline lib/ball.js]: Maybe 'stop' ?

File: lib/board.js [hunk 6]
```diff
@@ -33,6 +24,19 @@ Board.prototype.drawCubes = function(){
   });
 };
 
+Board.prototype.assignCubeParams = function(originX, originY, rowCount, cubeNumber){
```
Comment (s-espinosa) [inline lib/board.js:27]: Like this being pulled out. Had done some refactoring on naming here: https://github.com/s-espinosa/qbert/blob/refactor_and_test/lib/board.js

I'm fine with this as is, but let me know if you see anything you like.

File: lib/index.js [hunk 7]
```diff
@@ -28,26 +28,43 @@ initialsButton.addEventListener('click', function(){
 		'totalScore': $('#game-score').text()
 	};
 
-  var lsGames = localStorage.getItem('games') || '[]';
-  var games = JSON.parse(lsGames);
+  var standings = currentTopScores();
```
Comment (s-espinosa) [inline lib/index.js:31]: Love it.

File: lib/board.js (hunk 6 above)
Comment (kamiboers) [inline lib/board.js:27]: I haven't yet had a chance to look over your code, but I'm happy to discuss when we get together.

File: lib/ball.js (hunk 3 above)
Comment (kamiboers) [inline lib/ball.js]: Hm. Interestingly, I think that I made a mistake refactoring this, but it's still working, so... Let's take a (quick) closer look this morning.

**Comment by kamiboers** [comment]
I made most of the changes that you've suggested here 👍 and pushed them up to this PR, but we should discuss changes before merging this branch.

File: lib/ball.js (hunk 2 above)
Comment (kamiboers) [inline lib/ball.js]: ES6 says no '?' in function names. I think there's an ES5 workaround, but it looks kinda ugly.

**Pull request merged**